rfcs/0120-artifact-service.md

```diff
@@ -220,11 +230,80 @@ objsvc <-- s3 200 ETag {etag: 123456}
 uploader <-- objsvc 200
 ```
 
-### Deletetion
+### Part Size
+```
+GET /upload-info
+GET /upload-info/:origin
+```
+
+Multipart uploads in various backing services have differing requirements related
+to the size of each individual part.  This endpoint will return information about
+part sizes for a given origin.  In the case that the origin is not specified, the
+origin of the requester will be used.
+
+The response of this method for an S3 origin will be:
+
+```
+[
+  {
+    type: 'single-part',
+    max_size: 5 * 1024 * 1024,
+    min_size: 0,
+    max_part_count: 1,
+    min_part_count: 1,
+    min_part_size: 5 * 1024 * 1024,
+    max_part_size: 5 * 1024 * 1024,
+    part_size_multiple: 1,
+  },
+  {
+    type: 'multipart',
+    max_size: 5 * 1024 * 1024 * 1024,
+    min_size: 0,
+    max_part_count: 10000,
+    min_part_count: 1,
+    min_part_size: 5 * 1024 * 1024,
+    max_part_size: 5 * 1024 * 1024,
+    part_size_multiple: 1,
+  },
+]
+```
+
+All services will use a shape like this, changing their values to reflect their
+reality.  This format will give the uploader the ability to choose from the
+supported types of uploads.  Only those upload types which work with this
+upload flow will be supported, regardless of whether more options are available
+in the underlying API.
+
+The keys for this response will have the following meaning:
+
+* `type`: an identifier for the style of upload, unique to each underlying
+  service
```

Comment:
As an uploader, how should I interpret the array of requirement objects?  There's no `type` parameter to the create endpoint, so I'm not sure how I would choose between the two entries in the example.

Reply:
@jonasfj and I had a meeting to talk about this endpoint and we think we have a solution which avoids needing to have this endpoint at all.  Let me know if you think that proposal addresses your concerns.